### ISSUE TITLE
Non-ESM friendly dependencies cause build errors for Calcite consumers

### ISSUE THREAD

**Issue by maxpatiiuk** [Esri/calcite-components#10761]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

Calcite includes two dependencies that do not have an ESM build:
- [color](https://www.npmjs.com/package/color)
- [interactjs](https://www.npmjs.com/package/interactjs)

In Stencil, all dependencies were bundled in, so the bundler would convert them to ESM.
In Lumina, dependencies are externalized, but devDependencies are bundled in (this has benefits of smaller final app bundle size, [ability to apply security patches](https://github.com/Esri/calcite-design-system/issues/10760), among other things)

The non-ESM dependencies are causing issues for downstream consumers of Calcite:
- JS API's ESM->AMD build is failing on Calcite's non-ESM dependencies
- JS API's Angular samples have build errors
- Makes using code without a bundler harder

### Expected Behavior

- Update/replace/fork/drop the dependencies that don't have an ESM build
- OR, declare these as devDependencies, to make them bundled in, matching behavior of Stencil

### Reproduction Sample

Run "npm ci:dist" in JS API on the calcite-next-3.0.0 branch

### Reproduction Steps

Errors during JS API build:
```sh
Running "createCalciteComponentsAMD" task
node_modules/interactjs/dist/interact.min.js (3:203): The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
Warning: node_modules/@esri/calcite-components/dist/chunks/utils4.js (6:7): "default" is not exported by "node_modules/color/index.js", imported by "node_modules/@esri/calcite-components/dist/chunks/utils4.js". Use --force to continue.

node_modules/interactjs/dist/interact.min.js (3:203): The 'this' keyword is equivalent to 'undefined' at the top level of an ES module, and has been rewritten
Warning: node_modules/@esri/calcite-components/dist/components/calcite-dialog/customElement.js (3:7): "default" is not exported by "node_modules/interactjs/dist/interact.min.js", imported by "node_modules/@esri/calcite-components/dist/components/calcite-dialog/customElement.js". Use --force to continue.

```

### Reproduction Version

3.0.0-next.12

### Regression?

3.0.0-next.7

### Priority impact

impact - p1 - need for current milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/calcite-ui-icons
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Maps SDK for JavaScript

**Comment by maxpatiiuk** [comment]
Per information from Dasa (JS API team), they are generating "SBOM" based on dependencies used at runtime:
> So SBOM is software bill of materials” and should include all of the dependencies used at runtime. Esri has started supporting giving these to customers for some products.

The tool they are using only looks at dependencies, not devDependencies:

 > I've given an SBOM to the security team by using `npm sbom --sbom-format spdx --omit dev > arcgis-js-api.json` which only reports the regular deps. This is why I don't move the non-ESM ones to dev deps.

Thus, Calcite could keep `color` and `interactjs` in the `dependencies` array, but tell Lumina to externalize them using this config option:

```ts
    useLumina({
      build: {
        dependencies: {
          // Workaround for https://github.com/Esri/calcite-design-system/issues/10761
          externalize: ["color", "interactjs"],
        },
```

**Comment by jcfranco** [comment]
@maxpatiiuk Could you clarify something? If non-ESM dependencies are automatically externalized, what’s the purpose of explicitly externalizing them with the snippet above? How does applying it change the behavior?

**Comment by maxpatiiuk** [comment]
Ups, I mispoke in my last comment:

> Thus, Calcite could keep color and interactjs in the dependencies array, but tell Lumina to externalize them using this config option:

Updated the comment. I meant to say "bundleIn" not "externalize". Too much context switching 😊

---


For future reference:
- "devDependencies" are bundled
- "dependencies" and "peerDependencies" are automatically externalized.
- This behavior can be customized using the build.dependencies option.

I suggested moving non-ESM dependencies to devDependencies so that they get bundled in. Since that may be an issue for JS API's SBOM, we can keep these as dependencies, but bundle them in using the build.dependencies.bundleIn config.

**Comment by jcfranco** [comment]
Awesome, thanks for clarifying!

**Comment by geospatialem** [comment]
@maxpatiiuk Can you verify the above fix mitigates the build errors the JS API repo with `3.0.0-next.19`?

**Comment by maxpatiiuk** [comment]
It looks good during local build, but the NPM deployment doesn't look correct - https://unpkg.com/browse/@esri/calcite-components@3.0.0-next.19/dist/chunks/utils4.js

Last time I heard, @jcfranco and @benelan were debugging the GitHub actions pipeline to diagnose the issue.

**Comment by jcfranco** [comment]
This should be fixed by https://github.com/Esri/calcite-design-system/pull/10848.

**Comment by geospatialem** [comment]
@maxpatiiuk With the fix above, can you re-test with `3.0.0-next.26` to ensure the above is verified?

**Comment by DitwanP** [comment]
@geospatialem looks like Max is OOO I confirmed with Franco that we are now getting the output we would be expecting to see.

🍠 Verified on `3.0.0-next.26`
